docs/source/advanced/solvePNP.rst

```diff
@@ -5,14 +5,97 @@ SolvePNP
    :alt: Example view of solvePNP with a 2020 FRC vision target.
 
 Before the 3D features are enabled, the camera must be calibrated. Do this from the Cameras tab, as illustrated `in this informational video <https://www.youtube.com/watch?v=gCnwO6idKq4>`_, using the provided chessboard image printed using your printer's "fit to page" mode and a pair of calipers or a ruler.
 This must be done per resolution that solvePNP is to be used with. 
 
+The reason we need to first callibrate the camera is because cameras distorts images to a certain degree. The types of distortion are:
+
+- Radical distortion
+
+   - Radical distortion results in straight lines to appear bent or curved and is the most common type.
+
+- Tangential distortion
+
+   - Tangential distortion results in objects appearing farther away or closer than they are in real life. This is due to the lens of the camera not being aligned with the plane of the image.
+
+By finding how much radical or tangential distortion there is, we can then undistort any image from the camera.
+In order to measure the amount of distortion, we can take pictures of an image with a known shape and see how much the camera distorts that image.
+We can then use that information to correct the distortion errors.
+
+For example, below you can see a chessboard that is undistorted. To the right, you can see a radical and tangential distortions
+of the same chessboard taken from a pinhole camera. 
+
+.. image:: /images/solvepnp/distortion.PNG
+   :alt: Calibration using the provided chessboard image.
+
+
+Before callibrating:
+
+- Make sure the chessboard is 8 x 8 with the length and width of EACH sqaure is the same and correctly inputted into Chameleon
+- Collect images from different areas, tilts and distances, tilts of +/- 45 degrees is recommended, however, rotating too much may result in inaccurate data 
+- Lighting: make sure no shadows are being cast onto the chessboard, and the chessboard is recieving even lighting
+- Make sure you see rainbow stripes from the corners of the chessboard before you take a snapshot.
+
 .. image:: /images/solvepnp/Thumb.PNG
    :alt: Calibration using the provided chessboard image.
 
+
+After you complete the callibration process, on the command line, you'll see:
+
+- Camera Matrix
+
+   - The camera matrix is a 3 by 3 matrix as show below
+   - .. image:: /images/solvepnp/camera_matrix.PNG
+   - **fx** and **fy** : These are the focal lengths of the camera expressed in pixels
+   - **cx** and **cy** : These are the principal points and refer to the image center 
+
+- Distortion Coefficents
+
+   - distortion coefficents are vectors of up to 8 parameters, formated like : **k1, k2, p1, p2 [, k3 [, k4, k5, k6]]**
+   - **k1, k2..** are radical distortion coefficents
+   - **p1, p2...** are tangential distortion coefficents
+   - The distortion do not change based off the camera resolution, so the distortion coefficents will be the same for 640 x 480 images and 320 x 240 images
```

Comment:
Does the `Camera Matrix` change based on camera resolution?

Also please indicate that this is an example, not an all-inclusive list.

Reply:
Yeah, camera matrices change proportionally to changes in resolution, added that in.